### PR TITLE
wrap `cfile`'s `fclose()` to avoid warning on latest glibc

### PR DESCRIPTION
**Pull request by spoonincode**
latest glibc adds a `__nonnull` attribute on `fclose()` which ends up causing a bunch of warnings,
```
libraries/libfc/include/fc/io/cfile.hpp:22:65: warning: ignoring attributes on template argument ‘int (*)(FILE*)’ [-Wignored-attributes]
   22 |    using unique_file = std::unique_ptr<FILE, decltype( &fclose )>;
```

Not really sure the best way of fixing this. Simply wrapping the call like this gets rid of the warning.

### CODE REVIEW

File: libraries/libfc/include/fc/io/cfile.hpp [hunk 1]
```diff
@@ -19,7 +19,11 @@
 namespace fc {
 
 namespace detail {
-   using unique_file = std::unique_ptr<FILE, decltype( &fclose )>;
+   static void close_file(FILE* f) {
```
Comment (heifner) [inline libraries/libfc/include/fc/io/cfile.hpp:22]: I believe `inline` would be better here than `static`.

**Comment by greg7mdp** [comment]
Why not just replace the `unique_file` definition with:
`using unique_file = std::unique_ptr<FILE, decltype( [](FILE *f) noexcept { (void)fclose(f); } )>;`
and not provide `&fclose` when constructing a unique_file?

File: libraries/libfc/include/fc/io/cfile.hpp (hunk 1 above)
Comment (spoonincode) [reply libraries/libfc/include/fc/io/cfile.hpp:22]: ah yep

**Comment by spoonincode** [comment]
> Why not just replace the `unique_file` definition with: `using unique_file = std::unique_ptr<FILE, decltype( [](FILE *f) noexcept { (void)fclose(f); } )>;` and not provide `&fclose` when constructing a unique_file?

That caused a different warning for me (that to be honest I didn't investigate)
```
libraries/libfc/include/fc/io/cfile.hpp:31:7: warning: ‘fc::cfile’ has a field ‘fc::detail::unique_file fc::cfile::_file’ whose type has internal linkage [-Wsubobject-linkage]
   31 | class cfile {
```

File: libraries/libfc/include/fc/io/cfile.hpp [hunk 2]
```diff
@@ -19,7 +19,11 @@
 namespace fc {
 
 namespace detail {
-   using unique_file = std::unique_ptr<FILE, decltype( &fclose )>;
+   inline void close_file(FILE* f) {
```
Comment (greg7mdp) [inline libraries/libfc/include/fc/io/cfile.hpp:22]:
```suggestion
   inline void close_file(FILE* f) noexcept {
```

**Review by greg7mdp — APPROVED** [review]

**Review by heifner — APPROVED** [review]

**Pull request merged**